webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.js

```diff
@@ -1,17 +1,24 @@
 import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import { Modal } from 'patternfly-react';
 import PropTypes from 'prop-types';
+import { SET_MODAL_CLOSED } from './ForemanModalConstants';
+import { selectModalStateById } from './ForemanModalSelectors';
 import ForemanModalHeader from './subcomponents/ForemanModalHeader';
 import ForemanModalFooter from './subcomponents/ForemanModalFooter';
 import ModalContext from './ForemanModalContext';
 import { extractModalNodes } from './helpers';
 
 const ForemanModal = props => {
-  const { isOpen, onClose, title, ...propsToPassDown } = props; // don't pass down those two as it causes PF problems
+  const { id, title, ...propsToPassDown } = props; // don't pass down those two as it causes PF problems
   // Extract header and footer from children, if provided
   const { headerChild, footerChild, otherChildren } = extractModalNodes(
     props.children
   );
+  const openState = useSelector(state => selectModalStateById(state, id));
+  const isOpen = false || (openState && openState.open);
+  const dispatch = useDispatch();
+  const onClose = () => dispatch({ type: SET_MODAL_CLOSED, payload: { id } });
```

Comment:
@amirfefer Thanks for bringing up the concern.  I hadn't really thought about it much, but after poking around it seems a lot of people in the JS community are talking about it.

Scroll down to the comments below this [Redux hooks article](https://dev.to/gsto/new-redux-hooks-a-before-and-after-comparison-are-they-better-loj) and there's some really good discussion, including someone who has the same concerns as you.  (I really like [this comment](https://dev.to/sschottler/comment/bmkn) and also want to call out [Dan Abramov's 2019 update](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0) to his 2015 article.

The gist seems to be that Hooks allow better separation of stateful logic and presentational logic (JSX), but on the other hand require closer coupling of other things.

Also today I checked out [React Testing Library](https://testing-library.com/docs/react-testing-library/intro), which is a replacement for Enzyme with a very different philosophy.  They're saying that instead of testing instances of React components, it makes more sense to look at the DOM elements that the components output, and then test by looking for text etc. on the screen, like a human would.  (Incidentally, this is also supposed to work better with Hooks than Enzyme does.)

As far as testing in Foreman, for now I'm proposing a solution with Enzyme in the comment below.

Reply:
Thanks for bringing this up! I think hooks are a great feature but still without best practices yet. I agree we need to change our conventions due to the evolving process, and the enzyme alternative looks totally interesting. 

However, IMHO React is going fast, and breaking our conventions over and over makes the code difficult to newcomers and increase our technical debt.
I do think we should evolve together with react, but changing our conventions should be a part of dedicated design, after a deeper understanding of the new Hooks and its best practices and  with a wide consensus (discourse thread would be a great start)

Until then I think something like [this](https://github.com/theforeman/foreman/pull/7105#discussion_r349004392) can work :)